kyo-core/shared/src/test/scala/kyoTest/KyoSpecAbstract.scala

```diff
@@ -0,0 +1,85 @@
+package kyoTest
+
+import kyo.*
+import kyo.KyoApp.Effects
+import scala.concurrent.Future
+import scala.concurrent.duration.Duration
+import scala.language.postfixOps
+import zio.EnvironmentTag
+import zio.Trace
+import zio.ZIO
+import zio.ZLayer
+import zio.internal.stacktracer.SourceLocation
+import zio.test.*
+
+abstract class KyoSpecAbstract[S] extends ZIOSpecAbstract:
+    override type Environment = Any
+    override val environmentTag: EnvironmentTag[Any] = EnvironmentTag[Environment]
+    override def bootstrap: ZLayer[Any, Any, Any]    = ZLayer.empty
+
+    // copied from `TestAnnotation` as it's `private[zio]`
+    // https://github.com/zio/zio/blob/series/2.x/test/shared/src/main/scala/zio/test/TestAnnotation.scala#L91-L92
+    private val trace: TestAnnotation[List[SourceLocation]] =
+        TestAnnotation("trace", List.empty, _ ++ _)
+
+    def spec: Spec[Environment, Any]
+
+    def run[In](v: => In < S)(using Flat[In < S]): ZIO[Environment, Throwable, In]
+
+    def test[In <: TestResult](label: String)(assertion: => In < S)(using
+        fl: Flat[In < S],
+        sl: SourceLocation,
+        tr: Trace
+    ): Spec[Any, Throwable] =
+        Spec.labeled(
+            label,
+            Spec
+                .test(ZTest(label, run(assertion)), TestAnnotationMap.empty)
+                .annotate(trace, sl :: Nil)
+        )
+
+    def suite[In](label: String)(specs: In*)(using
+        sc: SuiteConstructor[In],
+        sl: SourceLocation,
+        tr: Trace
+    ): Spec[sc.OutEnvironment, sc.OutError] =
+        zio.test.suite(label)(specs*)
+
+end KyoSpecAbstract
+
+abstract class KyoSpecDefault extends KyoSpecAbstract[KyoApp.Effects]:
+    final override def run[In](v: => In < KyoApp.Effects)(using Flat[In < KyoApp.Effects]): ZIO[Environment, Throwable, In] =
+        ZIO.fromFuture { implicit ec => IOs.run(KyoApp.runFiber(timeout)(IOs(v)).toFuture).map(_.get) }
```

Comment:
Maybe we can add a `private[kyo] def KyoApp.runToFuture` method.

Note, I am wrapping in `IOs` here to ensure suspension - not sure that's necessary.

Reply:
Maybe `KyoApp.runFuture`? I think it could be public as well, it seems useful. There's no need for the `IOs` suspension since the code handles `IOs` and is also suspended in ZIO.

Reply:
Sure - I was just hesitant to suggest a public function as it seems like it could be misused.